### DISCUSSION TITLE
根据《学位论文审核的格式要求说明》修改模板选项

### DISCUSSION THREAD

**Discussion by sikouhjw** [sikouhjw/gdutthesis#74] [Announcements]
由于《关于毕业研究生提交学位论文电子版的通知》[^1]中的附件《附件2：学位论文审核的格式要求说明》[^2]规定了一些要求，现来分析如何修改模板选项使得符合《附件2：学位论文审核的格式要求说明》的规定。

> 1.上传论文必须有PDF和Word两个版本，且都可以下载和打开。
> 2.Word版本对封面格式不做要求，能下载和打开即可。

以下是导师跟研究生处沟通后的原话：
1. 跟研究生处沟通过了，只需要 PDF；
2. 图书馆那边，研究生处也已告知了，学生不放心，可以跟图书馆再联系下。

> 3.PDF版本必须有彩色封面，全文不能有空白页，声明页要有作者和导师的亲笔签名（扫描件或影印件均可，但字迹要清晰）。

要求有三点：
1. PDF 里需要有封面，只需要指定选项 `style/cover = {true}`；
2. 不能有空白页，即 `\chapter` 不需要指定为在奇数页输出，只需要指定选项 `style/open = {true}`；
3. 声明页需要签名，只需要使用 `\gdutstatement*[〈文件名〉]` 替代 `\gdutstatement` 来插入声明页。

> 4.【论文信息】里“专业”和“研究方向”可以空白，其他是必填项。
> 5.“中文摘要”“英文摘要”必须填写，不能截图插入。

跟模板无关。

> 6.中英文扉页中的“分类号”、“UDC”和“密级”可以不填，但是中英文扉页不能添加页眉、页眉线、页码或“注”。

根据《广东工业大学研究生学位论文撰写规范》[^3]：
> 注：属保密的论文，需办理相关手续后，注明“密级：”。“分类号”和“UDC”由档案馆填写。

密级可以通过修改 `info/secret-level = {none}` 来选择密级。

> 7.PDF文档不能出现空白页，也不能在空白页上添加页眉和页码。
> 一篇学位论文按顺序必须要有：封面（彩色）、中文扉页、英文扉页、中文摘要、英文摘要、中文目录、英文目录、正文、结论、参考文献、攻读学位期间发表的论文（专利）、学位论文独创性声明、学位论文版权使用授权声明（两个声明放在一页）、致谢、（附录）。

同“3.”。

> 8.页码问题：从”中文摘要”开始，到”英文目录”结束，以罗马大写数字排序，为I、II、III、IV、V、VI、VII、VIII、IX、X、XI、XII等，中间不能缺页，页码不能中断或重复。从正文第一章绪论开始，直至最后的致谢和附录，用阿拉伯数字排序，中间不能缺页，页码不能中断或重复。
> 9.页眉问题：奇数页的页眉应为当前章节名，如“第一章 绪论”， 不能只写“绪论”而漏掉章节，学术型硕士论文偶数页的页眉应为“广东工业大学硕士学位论文”，专业型硕士的论文偶数页的页眉应为“广东工业大学硕士专业学位论文”，全文如此（包括摘要、目录、声明、致谢、附录等页）。
> 10.参考文献应该放在正文之后，声明页之前，不能分散放在每章节之后。附录放在全文最后，即致谢页之后。

模板默认如此。

[^1]: https://oas.gdut.edu.cn/seeyon/newsData.do?method=newsView&newsId=2164583378238849893&from=&spaceId=
[^2]: https://oas.gdut.edu.cn/seeyon/fileDownload.do?method=download&fileId=2211376378435552953&v=0cd9e9c3d65a11721d8b5fc997845221&createDate=2022-06-14&filename=%E9%99%84%E4%BB%B62%EF%BC%9A%E5%AD%A6%E4%BD%8D%E8%AE%BA%E6%96%87%E5%AE%A1%E6%A0%B8%E7%9A%84%E6%A0%BC%E5%BC%8F%E8%A6%81%E6%B1%82%E8%AF%B4%E6%98%8E.docx
[^3]: https://yjs.gdut.edu.cn/system/_content/download.jsp?urltype=news.DownloadAttachUrl&owner=1533825352&wbfileid=8020315

**Comment by iTomxy** [comment]
1. 在指定 `style/open = {true}` 之后，一开封面（`sytle/cover = {true}`），我的会在英文扉页后面多一页空白页，要手动删。

2. 声明页的电子签名可以在 [gdutthesis.cls](https://github.com/sikouhjw/gdutthesis/blob/main/gdutthesis.cls) 里通过 minipage 硬编码进去（日期也是）。参照 [1]，在把电子签名整成 pdf、裁剪好之后：
    - 在 [1317](https://github.com/sikouhjw/gdutthesis/blob/main/gdutthesis.cls#L1317) 行和 [1324](https://github.com/sikouhjw/gdutthesis/blob/main/gdutthesis.cls#L1324) 左右，把原本的
    ```latex
    \begingroup \centering \hspace*{5\ccwd}论文作者签名：\hspace*{5\ccwd}日期：\par \endgroup
    ```
    改为
    ```latex
    \begingroup \centering \hspace*{5\ccwd}论文作者签名：\begin{minipage}{4.7\ccwd}
    	\includegraphics[width=4.5\ccwd]{my-signature.pdf}
    \end{minipage} 日期： 2022.5.25\par \endgroup
    ```

    - 在 [1326](https://github.com/sikouhjw/gdutthesis/blob/main/gdutthesis.cls#L1326) 行左右，把
    ```latex
    \begingroup \centering \hspace*{5\ccwd}指导教师签名：\hspace*{5\ccwd}日期：\par \endgroup
    ```
    改为
    ```latex
    \begingroup \centering \hspace*{5\ccwd}指导教师签名：\begin{minipage}{4.7\ccwd}
    	\includegraphics[width=4.5\ccwd]{supervisor-signature.pdf}
    \end{minipage} 日期： 2022.5.26\par \endgroup
    ```

3. 模板中声明页的英文标题是不是要改成首字母大写？即 [1305](https://github.com/sikouhjw/gdutthesis/blob/main/gdutthesis.cls#L1305) 和 [1312](https://github.com/sikouhjw/gdutthesis/blob/main/gdutthesis.cls#L1312) 行左右的
    ```latex
    Statement~of~original~authorship~and~copyright~licensing~declaration
    ```
    改成
    ```
    Statement~of~Original~Authorship~and~Copyright~Licensing~Declaration
    ```

---

1. [如何在签名处插入签名图片](https://ask.latexstudio.net/ask/question/7450.html)

**Reply by sikouhjw** [reply]
1. 这个是 bug，我之前没注意到，将
   https://github.com/sikouhjw/gdutthesis/blob/164512710fa782838b35d90c1546576556506f41/gdutthesis.cls#L1891
   改为 `\clearpage` 就行了
2. 这个有道理，可以给 `\gdutstatement` 加个可选参数：`\gdutstatement[日期][作者签名][导师签名]`
3. 这个大小写问题，我是参考《广东工业大学研究生学位论文格式模板-2022年5月》[^1]及它的前一个版本去设置的，Word 里写的的就是“Statement of original authorship and copyright licensing declaration”，包括其它的也是只有第一个单词的首字母大写：“Conclusion and prospect”、“Publication and patents during study”、“Background and significance of research”

[^1]: https://yjs.gdut.edu.cn/guangdonggongyedaxueyanjiushengxueweilunwengeshimoban-2022nian5yue.docx

**Comment by iTomxy** [comment]
根据提交电子版的管理员审核意见，[成果页的中文标题](https://github.com/sikouhjw/gdutthesis/blob/main/main.tex#L215)要改成「攻读学位期间取得与学位相关的成果」：
![pub-issue](https://user-images.githubusercontent.com/44149820/174052435-1909eb25-e6f7-43d2-b4d9-3769774bf42f.png)

**Reply by sikouhjw** [reply]
在《广东工业大学研究生学位论文格式模板-2022年5月》[^1]中，确确实实是『攻读学位期间取得与学位论文相关的成果』：
![图片](https://user-images.githubusercontent.com/49770925/174061245-babed350-85c2-4402-adfb-289d697337db.png)

看看是普遍现象还是个别现象，如果是个别现象就不改了。

PS：还有其它打回来的问题吗？

[^1]: https://yjs.gdut.edu.cn/guangdonggongyedaxueyanjiushengxueweilunwengeshimoban-2022nian5yue.docx

**Reply by iTomxy** [reply]
没有，改了这个就过了

**Comment by sikouhjw** [comment]
针对上传电子版学位论文需要将封面分别置于论文头和尾的要求，在此给出一个补丁：
1. 在导言区添加以下代码
```tex
\ExplSyntaxOn
\makeatletter
\cs_set:Npn \gdutcover
  {
    \newgeometry{margin=0pt}
      \begingroup
      \centering
      \begin{minipage}[c][29.7cm]{20.78cm}%
        \centering
        \vspace*{2cm}
        \begingroup \centering\sffamily\zihao{2} \l__gdut_info_title_tl \par \endgroup
        \vspace*{3cm}
        \begingroup \centering\zihao{4}\bfseries
        \begin{tabular}{c>{\centering\arraybackslash}p{16\ccwd}}
          \multirow{4}*{
            \begin{CJKfilltwosides*}{0pt}
              作者姓名：\\
              导师姓名：\\
              学科（专业）或领域名称：\\
              论文答辩年月：
            \end{CJKfilltwosides*}
          } & \l__gdut_info_author_tl \\
          \cline{2-2}
          & \l__gdut_info_supervisor_name_tl
          \str_if_eq:NNTF \l__gdut_info_supervisor_three_tl {none}
          { \\ }
          {
            \quad \l__gdut_info_supervisor_two_name_tl \\
          }
          \cline{2-2}
          & \l__gdut_info_major_tl \\
          \cline{2-2}
          & \exp_args:No \coverdate{\l__gdut_info_date_tl} \\
          \cline{2-2}
        \end{tabular}
        \par \endgroup
      \end{minipage}%
      \endgroup
    \restoregeometry
    \newpage
  }
\clist_const:Nn \c__gdut_background_new_clist
  {
    \backgroundsetup{
      scale     = 1,
      position  = {-2.78cm, 2.5cm},
      nodeanchor= north,
      contents  = { \includegraphics[width=42.5cm,height=29.7cm]{cover-background/doctor.jpeg}    }
      },
    \backgroundsetup{
      scale     = 1,
      position  = {-2.78cm, 2.5cm},
      nodeanchor= north,
      contents  = { \includegraphics[width=42.5cm,height=29.7cm]{cover-background/master.jpeg}    }
      },
    \backgroundsetup{
      scale     = 1,
      position  = {-2.78cm, 2.5cm},
      nodeanchor= north,
      contents  = { \includegraphics[width=42.5cm,height=29.7cm]{cover-background/promaster.jpeg} }
      }
  }
\cs_set_protected:Npn \__gdut_background_type:
  {
    \bool_if:NTF \l__gdut_cover_bool
    {
      \clist_item:Nn \c__gdut_background_new_clist
      { \g__gdut_thesis_type_int }
    }
    {
      \backgroundsetup{ contents = { } }
    }
  }
\cs_set:Npn \gdutendcover
  {
    \__gdut_end_background_type:
    \BgThispage
  }
\clist_const:Nn \c__gdut_background_end_clist
  {
    \backgroundsetup{
      scale     = 1,
      position  = {21cm-2.35cm, 2.5cm},
      nodeanchor= north,
      contents  = { \includegraphics[width=42.5cm,height=29.7cm]{cover-background/doctor.jpeg}    }
      },
    \backgroundsetup{
      scale     = 1,
      position  = {21cm-2.35cm, 2.5cm},
      nodeanchor= north,
      contents  = { \includegraphics[width=42.5cm,height=29.7cm]{cover-background/master.jpeg}    }
      },
    \backgroundsetup{
      scale     = 1,
      position  = {21cm-2.35cm, 2.5cm},
      nodeanchor= north,
      contents  = { \includegraphics[width=42.5cm,height=29.7cm]{cover-background/promaster.jpeg} }
      }
  }
\cs_set_protected:Npn \__gdut_end_background_type:
  {
    \clist_item:Nn \c__gdut_background_end_clist
    { \g__gdut_thesis_type_int }
  }
\makeatother
\ExplSyntaxOff
```
2. 在 `\input{data/appendix.tex}` 后、`\end{document}` 前的位置，添加
```tex
\clearpage\mbox{}\pagestyle{empty}
\gdutendcover
```

**Reply by JohnnyLin12** [reply]
作者你好，我按照上面代码希望封面前后显示，前面的封面页正常显示在开头，但是后面的封面跟我附录的表格内容显示在统一页了
![image](https://github.com/user-attachments/assets/2b40d9e5-7cd9-4452-8cef-f8476f87b559)，请问如何处理？

**Reply by sikouhjw** [reply]
> 作者你好，我按照上面代码希望封面前后显示，前面的封面页正常显示在开头，但是后面的封面跟我附录的表格内容显示在统一页了 !请问如何处理？

改成
```tex
\clearpage\mbox{}\pagestyle{empty}
\gdutendcover
```
试试？

**Reply by JohnnyLin12** [reply]
可行，十分感谢作者，回复太速度了（夸！）。

**Reply by Unclechan97** [reply]
作者你好！想问一下导言区是什么地方呢？问了几个同学也不清楚

**Reply by sikouhjw** [reply]
> 作者你好！想问一下导言区是什么地方呢？问了几个同学也不清楚

https://github.com/sikouhjw/gdutthesis/blob/0f3745127922c301296a210dcdcd5de119c00dce/main.tex#L112-L116
114-115 行就是导言区

**Reply by Unclechan97** [reply]
感谢作者！很棒的模板！:+1:

**Comment by yeung8991** [comment]
根据最新的规范要求，专硕的页眉和扉页和学硕保持一致，用“广东工业大学硕士学位论文”即可～可以改gdutthesis.cls

**Reply by sikouhjw** [reply]
下次一起修改

**Reply by sikouhjw** [reply]
在本地改了还没提交，等有空再更新模板

**Reply by sikouhjw** [reply]
https://github.com/sikouhjw/gdutthesis/blob/0f3745127922c301296a210dcdcd5de119c00dce/gdutthesis.cls#L169-L171
现在需要修改的话，修改这里
